lib/ssl/src/inet_tls_dist.erl

```diff
@@ -582,8 +574,8 @@ do_setup_connect(Driver, Kernel, Node, Address, Ip, TcpPort, Version, Type, MyNo
     Opts =  trace(connect_options(get_ssl_options(client))),
     dist_util:reset_timer(Timer),
     case ssl:connect(
-        Address, TcpPort,
-        [binary, {active, false}, {packet, 4},
+        Ip, TcpPort,
+        [binary, {active, false}, {packet, 4}, {server_name_indication, Address},
```

Comment:
why this adjustment is needed?

Reply:
It is needed because erl_epmd:address_please/3 will return an IP for the address it is given and the original address does not have to have a valid DNS record.